src/Orleans.Runtime/Storage/StateStorageBridge.cs

```diff
@@ -145,5 +160,13 @@ private string MakeErrorMsg(string what, Exception exc)
             return string.Format("Error from storage provider {0} during {1} for grain Type={2} Pk={3} Id={4} Error={5}" + Environment.NewLine + " {6}",
                 $"{this.store.GetType().Name}.{this.name}", what, name, grainRef.GrainId.ToDetailedString(), grainRef, errorCode, LogFormatter.PrintException(exc));
         }
+
+        private static void CheckRuntimeContext()
```

Comment:
I'd put [MethodImpl(MethodImplOptions.AggressiveInlining)] on this method.

Reply:
The method probably shouldn't be inlined as-is: first we should split the throw into a different method.
Submit a PR? 😛